tests/environment.cc

```diff
@@ -882,6 +870,43 @@ test_createLockfile_both()
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+/** @brief createLockfile gives a helpful error message when a package can't be
+ * resolved. */
+bool
+test_createLockfile_error()
+{
+  /* Create manifest with hello */
+  ManifestRaw manifestRaw;
+  manifestRaw.install          = { { "not-a-package", std::nullopt } };
+  manifestRaw.options          = Options {};
+  manifestRaw.options->systems = { _system };
+  manifestRaw.registry         = registryWithNixpkgs;
+
+  EnvironmentManifest manifest( manifestRaw );
+
+  /* Test locking manifest creates expectedLockfile */
+  Environment environment( std::nullopt, manifest, std::nullopt );
+  try
+    {
+      Lockfile actualLockfile = environment.createLockfile();
+      return false;
+    }
+  catch ( flox::FloxException & err )
+    {
+      EXPECT_EQ(
+        err.what(),
+        std::string(
+          "resolution failure: failed to resolve some package(s):\n"
+          "  in default group:\n"
+          "    failed to resolve `not-a-package' in input "
+          "`github:NixOS/nixpkgs/e8039594435c68eb4f780f3e9bf3972a7399c4b1'" ) );
```

Comment:
@zmitchell do we need to tweak the newlines or indentation so this plays nicely with our error formatting in Rust, or does it already look okay as is?

Reply:
Good catch. It's kind of hard to tell whether a newline in the message here will inherit the indentation that the Rust side applies to each error in the chain of errors. Can you try setting `PKGDB_BIN` in flox at runtime to see how it looks?

Reply:
It seems context message is not being printed at all:
```
> PKGDB=./bin/pkgdb SHELL=bash flox --debug activate
...
[DEBUG] [posix] + locking manifest with command: "/Users/matthew/pkgdb/main/bin/pkgdb" "manifest" "lock" "--ga-registry" "/Users/matthew/debug/.flox/env/manifest.toml"
[ERROR] [flox] locking manifest failed: resolution failure
> "/Users/matthew/pkgdb/main/bin/pkgdb" "manifest" "lock" "--ga-registry" "/Users/matthew/debug/.flox/env/manifest.toml"
resolution failure: failed to resolve some package(s):
  in default group:
    failed to resolve `bad1' in input `github:NixOS/nixpkgs/f741f8a839912e272d7e87ccf4b9dbc6012cdaf9'
```

Reply:
Is this the issue? https://github.com/flox/flox/blob/93785d22b78e59e7a5dcaa4cdf1e18956c608e50/crates/flox-rust-sdk/src/models/pkgdb_errors.rs#L68

Reply:
I don't think that's the issue. If a type implementing `std::error::Error` implements the `source` method then `anyhow` will continue going down that chain of sources until it gets a `None` back. This line is just printing the message for this "link" in the chain.